staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/cbor.go

```diff
@@ -141,12 +143,75 @@ func (s *serializer) unmarshal(data []byte, into interface{}) (strict, lax error
 	if u, ok := into.(runtime.Unstructured); ok {
 		var content map[string]interface{}
 		defer func() {
-			// TODO: The UnstructuredList implementation of SetUnstructuredContent is
-			// not identical to what unstructuredJSONScheme does: (1) it retains the
-			// "items" key in its Object field, and (2) it does not infer a singular
-			// Kind from the list's Kind and populate omitted apiVersion/kind for all
-			// entries in Items.
-			u.SetUnstructuredContent(content)
+			switch u := u.(type) {
+			case *unstructured.UnstructuredList:
+				// UnstructuredList's implementation of SetUnstructuredContent
+				// produces different objects than those produced by a decode using
+				// UnstructuredJSONScheme:
+				//
+				//   1. SetUnstructuredContent retains the "items" key in the list's
+				//      Object field. It is omitted from Object when decoding with
+				//      UnstructuredJSONScheme.
+				//   2. SetUnstructuredContent does not populate "apiVersion" and
+				//      "kind" on each entry of its Items
+				//      field. UnstructuredJSONScheme does, inferring the singular
+				//      Kind from the list Kind.
+				//   3. SetUnstructuredContent ignores entries of "items" that are
+				//      not JSON objects or are objects without
+				//      "kind". UnstructuredJSONScheme returns an error in either
+				//      case.
+				//
+				// UnstructuredJSONScheme's behavior is replicated here.
+				var items []interface{}
+				if uncast, present := content["items"]; present {
+					var cast bool
+					items, cast = uncast.([]interface{})
+					if !cast {
+						strict, lax = nil, fmt.Errorf("items field of UnstructuredList must be encoded as an array or null if present")
+						return
```

Comment:
if someone is using lax, they'll get no error and also no content. What would someone do with that return?  is that a useful return?

Given this is a new serializer down an error case, do we have any viable use-case for not failing and also not returning a useable list?  Is it better to always fail?

Reply:
Seems like it's better to fail both lax and strict.  We can't make sense of how to decode in these cases.

Reply:
I will add an explanation about these errors in the comments. It does fail in both cases. The last return value is meant to be interpreted according to the usual Go convention, and indicates that the unmarshal operation failed entirely.

The signature is roughly based on [UnmarshalStrict](https://github.com/kubernetes-sigs/json/blob/bc3834ca7abd3a90f03ef00a27ad80cb892f9c21/json.go#L98) from `sigs.k8s.io/json`.